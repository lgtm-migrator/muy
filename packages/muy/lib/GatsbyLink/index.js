import React from "react"
import Link from "@material-ui/core/Link"
import { Link as GatsbyLinkComponent } from "gatsby"

const defaultProps = {
  target: "_self",
}

const GatsbyLink = ({ rel, target = defaultProps.target, to, ...props }) => (
  <Link
    component={GatsbyLinkComponent}
    rel={target === "_blank" ? `noopener noreferer ${rel}` : rel}
    target={target}
    to={to}
    {...props}
  />
)

export default GatsbyLink
