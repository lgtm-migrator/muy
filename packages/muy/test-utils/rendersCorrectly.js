import renderer from "react-test-renderer"

const rendersCorrectly = component => {
  const tree = renderer.create(component).toJSON()
  return expect(tree).toMatchSnapshot()
}

export default rendersCorrectly
