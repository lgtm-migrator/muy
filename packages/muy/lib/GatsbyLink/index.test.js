import React from "react"
import rendersCorrectly from "../../test-utils/rendersCorrectly"
import GatsbyLink from "."

describe("<GatsbyLink />", () => {
  it("renders correctly", () => {
    rendersCorrectly(<GatsbyLink to={"/"}>Link</GatsbyLink>)
  })
})
