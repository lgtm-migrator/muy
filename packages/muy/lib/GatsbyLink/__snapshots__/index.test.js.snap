// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GatsbyLink /> renders correctly 1`] = `
<a
  aria-current="page"
  className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
  href="/"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  style={Object {}}
  target="_self"
>
  Link
</a>
`;
